Dockerfile

```diff
@@ -1,50 +1,21 @@
-FROM rocker/r-ver:4.3.2
-
-# Set the working directory to setup. Uses a dedicated directory instead of
-# root since otherwise renv will try to scan every subdirectory
-WORKDIR /setup
-
-# Use PPM for binary installs
-ENV RENV_CONFIG_REPOS_OVERRIDE "https://packagemanager.posit.co/cran/__linux__/jammy/latest"
-ENV RENV_CONFIG_SANDBOX_ENABLED FALSE
-ENV RENV_PATHS_LIBRARY renv/library
-ENV RENV_PATHS_CACHE /setup/cache
+FROM pytorch/pytorch:2.3.0-cuda12.1-cudnn8-runtime
```

Comment:
Installing CUDA [appears to be a pain](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/), so for the purposes of benchmarking this PR I just switched to a base image that comes with CUDA installed and refactored the Dockerfile to only install dependencies necessary for running `python/comps.py`.